compendium/AsyncWorkService/include/cppmicroservices/asyncworkservice/AsyncWorkService.hpp

```diff
@@ -67,6 +67,12 @@ namespace cppmicroservices
              * with the std::packaged_task<void()> in order to wait on the async task.
              */
             virtual void post(std::packaged_task<void()>&& task) = 0;
+
+            /**
+             * Method to wait for all tasks posted to the service to finish
```

Comment:
Is it worth adding a note here that implementing this is recommended to avoid odd behaviour if an implementation of this interface is unregistered whilst it has unfinished work posted to it?

Reply:
right now this is a required method -- I think maybe this is worth discussing in CppMs Working group tomorrow